include/fiction/algorithms/simulation/sidb/is_operational.hpp

```diff
@@ -673,13 +681,138 @@ template <typename Lyt, typename TT>
+    }
+
+    const auto non_op_patterns_and_non_op_reason =
+        p.determine_non_operational_input_patterns_and_non_operationality_reason();
+
+    for (const auto& [input_pattern, _] : non_op_patterns_and_non_op_reason)
+    {
+        input_patterns.erase(input_pattern);
+    }
+
+    return input_patterns;
+}
+
+/**
+ * This function determines all input combinations for which kinks induce the SiDB layout to become non-operational.
+ * This means that the layout is operational if kinks would be accepted.
+ *
+ * @note "kink induced non-operational" refers to the operational status being exclusively caused to be
+ * `NON_OPERATIONAL` by kinks.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @tparam TT Type of the truth table.
+ * @param lyt The SiDB layout.
+ * @param spec Vector of truth table specifications.
+ * @param params Parameters for the `is_operational` algorithm.
+ * @return The input combinations where kinks induce the SiDB layout to become non-operational.
+ */
+template <typename Lyt, typename TT>
+[[nodiscard]] std::set<uint64_t>
+kink_induced_non_operational_input_patterns(const Lyt& lyt, const std::vector<TT>& spec,
+                                            const is_operational_params& params = {}) noexcept
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt is not an SiDB layout");
+    static_assert(kitty::is_truth_table<TT>::value, "TT is not a truth table");
+
+    assert(lyt.num_pis() > 0 && "skeleton needs input cells");
+    assert(lyt.num_pos() > 0 && "skeleton needs output cells");
+
+    assert(!spec.empty());
+    // all elements in tts must have the same number of variables
+    assert(std::adjacent_find(spec.cbegin(), spec.cend(), [](const auto& a, const auto& b)
+                              { return a.num_vars() != b.num_vars(); }) == spec.cend());
+
+    is_operational_params params_with_rejecting_kinks = params;
+    params_with_rejecting_kinks.op_condition          = operational_condition::REJECT_KINKS;
+
+    detail::is_operational_impl<Lyt, TT> p{lyt, spec, params_with_rejecting_kinks};
+
+    std::set<uint64_t> kink_induced_non_op_patterns{};
+
+    const auto input_patterns_and_non_op_reason =
+        p.determine_non_operational_input_patterns_and_non_operationality_reason();
+
+    for (const auto& [input_pattern, status] : input_patterns_and_non_op_reason)
+    {
+        if (status == detail::non_operationality_reason::KINKS)
+        {
+            kink_induced_non_op_patterns.insert(input_pattern);
+        }
+    }
+
+    return kink_induced_non_op_patterns;
+}
+
+/**
+ * This function determines if the layout is only considered as non-operational because of kinks. This means that
+ * the layout would be considered as operational, if kinks were accepted.
+ *
+ * @note "Kink induced non-operational" refers to the operational status being exclusively caused to be
+ * `NON_OPERATIONAL` due to kinks.
```

Comment:
```suggestion
 * @note "Kink induced non-operational" refers to the non-operational status being exclusively caused by kinks with an otherwise correct logic match.
```